packages/common/src/utils/shallowEqual.ts

```diff
@@ -0,0 +1,43 @@
+/*
```

Comment:
I think these additions (useContextSelector and shallowEqual files) would be better as a different PR.